buildSrc/src/main/groovy/org/elasticsearch/gradle/test/DistroTestPlugin.java

```diff
@@ -164,9 +177,30 @@ private void configureCopyPackagingTask(Project project) {
         this.copyPackagingArchives = Boilerplate.maybeRegister(project.getParent().getTasks(), COPY_PACKAGING_TASK, Copy.class,
             t -> {
                 t.into(archivesDir);
-                t.from(project.getConfigurations().getByName(PACKAGING_DISTRIBUTION));
+                t.from(project.getConfigurations().getByName(PACKAGING_CONFIGURATION));
 
                 Path archivesPath = archivesDir.get().getAsFile().toPath();
+                TaskInputs inputs = t.getInputs();
+                inputs.property("version", VersionProperties.getElasticsearch());
+                t.doLast(action -> {
+                    try {
+                        Files.writeString(archivesPath.resolve("version"), VersionProperties.getElasticsearch());
+                    } catch (IOException e) {
+                        throw new UncheckedIOException(e);
+                    }
+                });
+        });
+    }
+
+    private void configureCopyUpgradeTask(Project project) {
+        this.upgradeDir = project.getParent().getLayout().getBuildDirectory().dir("packaging/upgrade");
+        // temporary, until we have tasks per distribution
+        this.copyUpgradeArchives = Boilerplate.maybeRegister(project.getParent().getTasks(), COPY_UPGRADE_TASK, Copy.class,
+            t -> {
+                t.into(upgradeDir);
+                t.from(project.getConfigurations().getByName(UPGRADE_CONFIGURATION));
```

Comment:
Something doesn't seem quite right here. We create this configuration per-project, but we only every create a single one of these tasks. Essentially, the first distro test project to be configured "wins" here. Is that the intention?

Reply:
Sorry this is a bug. The intention was to have the configuration on the parent (qa/vagrant).